org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/GuardedPattern.java

```diff
@@ -260,4 +265,30 @@ public void setPattern(Pattern pattern) {
 		postReplaceChild(oldChild, pattern, PATTERN_PROPERTY);
 	}
 
+	/**
+	 * A character index into the original restricted identifier source string, or <code>-1</code> if no restricted
+	 * identifier source position information is available for this node; <code>-1</code> by default.
+	 * @noreference
+	 * since 3.30
+	 */
+	public void setRestrictedIdentifierStartPosition(int restrictedIdentifierStartPosition) {
```

Comment:
Should this method be public or protected?
Only supposed to be called by the ASTConverter ?

We need this value to be copied across in clone method also.